mavros_extras/src/plugins/debug.cpp

```diff
@@ -0,0 +1,254 @@
+
+		// subscribers
+		debug_sub = debug_nh.subscribe("send", 10, &DebugPlugin::debug_cb, this);
+
+		debug_nh.param("publish_debug", publish_debug, false);
+
+		// publishers (optional)
+		if (publish_debug) {
+			debug_pub = debug_nh.advertise<mavros_msgs::Debug>("debug", 10);
+			/** @todo create a pub config based on the naming of the debug msgs,
+			 * using a map_dict as in distance_sensor plugin
+			 */
+			debug_vector_pub = debug_nh.advertise<mavros_msgs::Debug>("debug_vector", 10);
+			named_value_float_pub = debug_nh.advertise<mavros_msgs::Debug>("named_value_float", 10);
+			named_value_int_pub = debug_nh.advertise<mavros_msgs::Debug>("named_value_int", 10);
+		}
+	}
+
+	Subscriptions get_subscriptions() {
+		return {
+			       make_handler(&DebugPlugin::handle_debug),
+			       make_handler(&DebugPlugin::handle_debug_vector),
+			       make_handler(&DebugPlugin::handle_named_value_float),
+			       make_handler(&DebugPlugin::handle_named_value_int)
+		};
+	}
+
+private:
+	ros::NodeHandle debug_nh;
+
+	ros::Subscriber debug_sub;
+
+	ros::Publisher debug_pub;
+	ros::Publisher debug_vector_pub;
+	ros::Publisher named_value_float_pub;
+	ros::Publisher named_value_int_pub;
+
+	// decide if we publish the debug data to topics or not
+	bool publish_debug;
+
+	/* -*- message handlers -*- */
+
+	/**
+	 * @brief Handle DEBUG message.
+	 * Message specification: http://mavlink.org/messages/common/#DEBUG
+	 * @param msg	Received Mavlink msg
+	 * @param debug	DEBUG msg
+	 */
+	void handle_debug(const mavlink::mavlink_message_t *msg, mavlink::common::msg::DEBUG &debug)
+	{
+		auto debug_msg = boost::make_shared<mavros_msgs::Debug>();
+
+		ros::Time stamp = m_uas->synchronise_stamp(debug.time_boot_ms);
+
+		ROS_DEBUG_STREAM_NAMED("debug", "timestamp: " << stamp
+							      << " type: DEBUG (indexed)"
+							      << " index: " << debug.ind
+							      << " value: " << debug.value);
+
+		debug_msg->header.stamp = stamp;
+		debug_msg->type = debug_msg->DEBUG_TYPE;
+		debug_msg->index = debug.ind;
+		debug_msg->value_float = debug.value;
+
+		debug_pub.publish(debug_msg);
+	}
+
+	/**
+	 * @brief Handle DEBUG_VECTOR message.
+	 * Message specification: http://mavlink.org/messages/common/#DEBUG_VECTOR
+	 * @param msg	Received Mavlink msg
+	 * @param debug	DEBUG_VECTOR msg
+	 */
+	void handle_debug_vector(const mavlink::mavlink_message_t *msg, mavlink::common::msg::DEBUG_VECT &debug)
+	{
+		auto debug_msg = boost::make_shared<mavros_msgs::Debug>();
+
+		std::string name(debug.name.data());
+		ros::Time stamp = m_uas->synchronise_stamp(debug.time_usec);
+		auto vector = Eigen::Vector3d(debug.x, debug.y, debug.z);
+
+		ROS_DEBUG_STREAM_NAMED(name, "timestamp: " << stamp
+							   << " type: DEBUG_VECTOR"
+							   << " vector: " << vector);
+
+		if (publish_debug) {
+			debug_msg->header.stamp = stamp;
+			debug_msg->name = name;
+			debug_msg->type = debug_msg->DEBUG_VECT_TYPE;
+			tf::vectorEigenToMsg(vector, debug_msg->vector);
```

Comment:
I don't think that eigen needed here. Maybe cog.py...

Reply:
well the statement is much simpler like this, don't you think?